src/redux/slices/userSlice.js

```diff
@@ -2,26 +2,59 @@ import { createSlice } from "@reduxjs/toolkit"
 
 import history from "lib/utils/history"
 import userAPI from "lib/api/user"
+import { DEFAULT_USER_IMAGE } from "lib/utils/constants"
 
 const userSlice = createSlice({
   name: "user",
-  initialState: { token: "" },
+  initialState: {
+    isLoading: false,
+    isError: false,
+  },
   reducers: {
     setCurrentUser(state, action) {
-      state.token = action.payload.token
+      for (const x in action.payload) {
+        state[x] = x === "image" ? DEFAULT_USER_IMAGE : action.payload[x]
+      }
```

Comment:
To jest nie bardzo. Lepiej stworz sobie w initialState usera ze wszystkimi polami i normalnie je przypisz, np.
state.user.username = action.payload.username